lib/house.rb

```diff
@@ -0,0 +1,58 @@
+class House
+  attr_reader :price,
+              :address,
+              :rooms,
+              :house_details
+  def initialize(price, address)
+    @price = price
+    @address = address
+    @rooms = []
+    @house_details = Hash.new
+  end
+
+  def price
+    @price.delete("$").to_i
+  end
+
+  def add_room(room)
+    @rooms << room
+  end
+
+  def above_market_average?
+    price > 500000
+  end
+
+  def rooms_from_category(category)
+    category_of_rooms = []
+    rooms.each do |room|
+      (category_of_rooms << room) if (room.category == category)
+    end
+    category_of_rooms
+  end
+
+  def house_area
+    rooms.map do |room|
+      room.area
+    end
+  end
+
+  def area
+    house_area.sum
```

Comment:
This makes sense in conjuction with the `house_area` method on line 33.  You can also use the ruby enumerable `reduce` here as well and that way we can just have one method that calculates area for a house